### ISSUE TITLE
Loading with 'async' can break filter

### ISSUE THREAD

**Issue by JonnyBurger** [LeaVerou/stretchy#20]
Elements get matched by `data-filter` even though they shouldn't if you load stretchy asynchronously.

Example: http://codepen.io/JonnyBurger/pen/rOEPOV
Same example, but without `async`: http://codepen.io/JonnyBurger/pen/ZbdwWY here everything is ok.

What happens is the following: 
1. Stretchy gets initialized, has default filter '*'
2. Elements get added to the page, Stretchy listens to them.
3. DOMContentLoaded, Stretchy updates to user defined filter, but previous elements are already watched.

I will follow up with a pull request shortly.

**Comment by JonnyBurger** [comment]
Wow, after some debugging I've realized the reason is another one! 
If you load the `<script>` with async, and there is a script after that, `_.script` becomes the last script in the document! Incidentially, CodePen also injects a script after my pen so that's why I was able to show the bug. 

I have now replaced `$$("script")` with `document.currentScript || $("script")`. This doesn't work in IE, but at least the bug will disappear in modern browsers. And it's a bit faster.
